release: getting blockers for release branch

celiala opened the pull request.

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/75514)
<!-- Reviewable:end -->

celiala committed:
release: refactor logic for getting required os env vars

pickSHA and getReleaseBlockers require different os env vars.

This commit refactors out logic for getting required os env vars,
so that we can customize which env vars are required for each
command.

celiala commented:
closing in favor of https://github.com/cockroachdb/cockroach/pull/76603
